components/avatar/avatar.vue

```diff
@@ -252,105 +286,82 @@ export default {
+    fullName: {
+      immediate: true,
+      handler (newName) {
+        this.formatInitials(newName);
+      },
+    },
+
+    imageSrc: {
+      immediate: true,
+      async handler (newSrc) {
+        if (!newSrc) return;
+
+        this.imageLoadedSuccessfully = null;
+        await this.$nextTick();
+        this.setImageListeners();
+      },
+    },
   },
 
   methods: {
-    async init () {
-      if (this.initializing) return;
-      this.kind = null;
-      await this.$nextTick();
-      const firstChild = this.$refs.canvas?.firstElementChild || this.$refs.canvas;
-      this.formatInitials(this.initials);
-      if (firstChild) {
-        this.setKind(firstChild);
-        this.kindHandler(firstChild);
-      }
-      this.initializing = true;
-      await this.$nextTick();
-      this.initializing = false;
-    },
+    setImageListeners () {
+      const el = this.$refs.avatarImage;
 
-    // eslint-disable-next-line complexity
-    kindHandler (el) {
-      switch (this.kind) {
-        case 'image':
-          el.classList.add('d-avatar__image');
-          this.validateImageAttrsPresence();
-          this.setImageListeners(el);
-          break;
-        case 'icon':
-          el.classList.add(AVATAR_KIND_MODIFIERS.icon);
-          break;
-        case 'initials':
-          this.slottedInitials = el.text || el.textContent;
-          this.formatInitials(this.slottedInitials.trim() || this.initials);
-          break;
-      }
-    },
+      if (!el) return;
 
-    setImageListeners (el) {
       el.addEventListener('load', () => this._loadedImageEventHandler(el), { once: true });
+
+      if (this.imageLoadedSuccessfully === false) return;
       el.addEventListener('error', () => this._erroredImageEventHandler(el), { once: true });
     },
 
-    formatInitials (initials) {
-      if (!initials || this.validatedSize === 'xs') {
+    formatInitials (string) {
+      const initials = this._extractInitials(string);
+
+      if (this.validatedSize === 'xs') {
         this.formattedInitials = '';
       } else if (this.validatedSize === 'sm') {
-        this.formattedInitials = initials.trim()[0];
+        this.formattedInitials = initials[0];
       } else {
-        this.formattedInitials = initials.trim().slice(0, 2);
+        this.formattedInitials = initials;
       }
     },
 
-    setKind (element) {
-      if (this.isIconType(element)) { this.kind = 'icon'; return; }
-      if (this.isImageType(element)) { this.kind = 'image'; return; }
-      this.kind = 'initials';
-    },
-
-    isIconType (element) {
-      return element?.tagName?.toUpperCase() === 'SVG';
-    },
-
-    isImageType (element) {
-      return element?.tagName?.toUpperCase() === 'IMG';
-    },
-
     getColor () {
       return this.color ?? getRandomElement(AVATAR_COLORS, this.seed);
     },
 
-    validateImageAttrsPresence () {
-      const isSrcMissing = !this.$refs.canvas.firstChild.getAttribute('src');
+    _extractInitials (string) {
```

Comment:
Could we move this `_extractInitials (string)` to an util function inside avatar folder?

Reply:
Code suggestion:
change `string` to `fullName`

do the trim direclty on the names constant and get the initials in a for loop

```
extractInitials (fullName) {
      if (typeof fullName !== 'string' || !fullName.trim()) return '';

      const names = fullName.trim().split(' ');
      let initials = '';

      for (const name of names) {
        initials += name.substring(0, 1).toUpperCase();
      }

      return initials;
    },
```

Reply:
Sure, I'll extract the method to an `utils.js` file inside avatar folder.

Thanks for the suggestion, improved the code.